### ISSUE TITLE
asm insturctions giving error on OSX 10.15.1

### ISSUE THREAD

**Issue by AravGarg** [Gallopsled/pwntools#1375]
Assembly instructions are giving an error that says:
Traceback (most recent call last):
  File "start.py", line 13, in <module>
    'mov eax, 0xb'
  File "/Users/arav/Library/Python/2.7/lib/python/site-packages/pwnlib/context/__init__.py", line 1440, in setter
    return function(*a, **kw)
  File "/Users/arav/Library/Python/2.7/lib/python/site-packages/pwnlib/asm.py", line 643, in asm
    assembler = _assembler()
  File "/Users/arav/Library/Python/2.7/lib/python/site-packages/pwnlib/asm.py", line 246, in _assembler
    version = re.search(r' (\d\.\d+)', result).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
[*] Closed connection to chall.pwnable.tw port 10000

Basically any assembly instruction is giving this 'NoneType' object has no attribute 'group' error.
Any fixes?

**Comment by Arusekk** [comment]
What is your gdb version? i.e. what output do you get from running `gdb --version`?

**Comment by k-moonblade** [comment]
i have the same error 

gdb --version                                                              
GNU gdb (GDB) 7.12

**Comment by frankli0324** [comment]
same error on

```
gdb --version
GNU gdb (GDB) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

**Comment by Arusekk** [comment]
The thing is that somehow the 9.2 does not get detected. I have totally no idea about what happens there. You could possibly add a debug `print(repr(result))` on the line before and post the result.

**Comment by thelongestusernameofall** [comment]
> The thing is that somehow the 9.2 does not get detected. I have totally no idea about what happens there. You could possibly add a debug `print(repr(result))` on the line before and post the result.

the result print like this: 

'Apple clang version 12.0.5 (clang-1205.0.22.11)\nTarget: x86_64-apple-darwin20.5.0\nThread model: posix\nInstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin\n'

**Comment by Arusekk** [comment]
Try to reproduce on latest dev version.
